mlflow/data/numpy_dataset.py

```diff
@@ -0,0 +1,148 @@
+        :param name: The name of the dataset. E.g. "wiki_train". If unspecified, a name is
+                     automatically generated.
+        :param digest: The digest (hash, fingerprint) of the dataset. If unspecified, a digest
+                       is automatically computed.
+        """
+        self._features = features
+        self._targets = targets
+        super().__init__(source=source, name=name, digest=digest)
+
+    def _compute_digest(self) -> str:
+        """
+        Computes a digest for the dataset. Called if the user doesn't supply
+        a digest when constructing the dataset.
+        """
+        return compute_numpy_digest(self._features, self._targets)
+
+    def _to_dict(self, base_dict: Dict[str, str]) -> Dict[str, str]:
+        """
+        :param base_dict: A string dictionary of base information about the
+                          dataset, including: name, digest, source, and source
+                          type.
+        :return: A string dictionary containing the following fields: name,
+                 digest, source, source type, schema (optional), profile
+                 (optional).
+        """
+        base_dict.update(
+            {
+                "schema": json.dumps({"mlflow_tensorspec": self.schema.to_dict()}),
+                "profile": json.dumps(self.profile),
+            }
+        )
+        return base_dict
+
+    @property
+    def source(self) -> DatasetSource:
+        """
+        The source of the dataset.
+        """
+        return self._source
+
+    @property
+    def features(self) -> Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]:
+        """
+        The features of the dataset.
+        """
+        return self._features
+
+    @property
+    def targets(self) -> Optional[Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]]:
+        """
+        The targets of the dataset. May be None if no targets are available.
+        """
+        return self._targets
+
+    @property
+    def profile(self) -> Optional[Any]:
+        """
+        A profile of the dataset. May be None if no profile is available.
+        """
+        return {
+            "shape": self._features.shape,
+            "size": self._features.size,
+            "nbytes": self._features.nbytes,
+        }
+
+    @cached_property
+    def schema(self) -> Optional[Schema]:
+        """
+        An MLflow TensorSpec schema representing the tensor dataset
+        """
+        try:
+            return _infer_schema(self._features)
+        except Exception as e:
+            _logger._warning("Failed to infer schema for Hugging Face dataset. Exception: %s", e)
+            return None
+
+    def to_pyfunc(self) -> PyFuncInputsOutputs:
+        """
+        Converts the dataset to a collection of pyfunc inputs and outputs for model
+        evaluation. Required for use with mlflow.evaluate().
+        May not be implemented by all datasets.
```

Comment:
Can we remove the "May not be implemented by all datasets" since it is implemented by all Numpy datasets?